### PR TITLE
feature/tensorrt engine op

### PR DESCRIPTION
**Pull request by Superjomn**

### CODE REVIEW

File: paddle/fluid/operators/tensorrt_engine_op.h [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    return kt;
+  }
+};
+
+template <typename DeviceContext, typename T>
+class TensorRTEngineKernel : public framework::OpKernel<T> {
+ public:
+  void Compute(const framework::ExecutionContext& context) const override {
+    if (!engine_) {
+      Prepare(context);
```
Comment (panyx0718) [inline paddle/fluid/operators/tensorrt_engine_op.h]: this seems not thread-safe?

File: paddle/fluid/operators/tensorrt_engine_op.h [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    PADDLE_ENFORCE_LE(batch_size, max_batch_);
+
+    // Convert input tensor from fluid to engine.
+    for (const auto& x : context.Inputs("Xs")) {
+      // convert input and copy to TRT engine's buffer
+      auto* v = context.scope().FindVar(x);
+      PADDLE_ENFORCE_NOT_NULL(v, "no variable called %s", x);
+      auto& t = v->Get<framework::LoDTensor>();
+      if (platform::is_cpu_place(t.place())) {
+        engine_->SetInputFromCPU(x, static_cast<const void*>(t.data<void*>()),
```
Comment (panyx0718) [inline paddle/fluid/operators/tensorrt_engine_op.h]: t.data<void> will return void*?

File: paddle/fluid/operators/tensorrt_engine_op.h [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                                 t.memory_size());
+      } else {
+        engine_->SetInputFromGPU(x, static_cast<const void*>(t.data<void*>()),
```
Comment (panyx0718) [inline paddle/fluid/operators/tensorrt_engine_op.h]: same here.

File: paddle/fluid/operators/tensorrt_engine_op.h [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+#ifdef PADDLE_WITH_CUDA
+
+#include "paddle/fluid/framework/operator.h"
+#include "paddle/fluid/inference/analysis/helper.h"
+#include "paddle/fluid/inference/tensorrt/engine.h"
+
+namespace paddle {
+namespace operators {
+
+class TensorRTEngineOp : public framework::OperatorWithKernel {
```
Comment (panyx0718) [inline paddle/fluid/operators/tensorrt_engine_op.h]: need a test for this op?

File: paddle/fluid/operators/tensorrt_engine_op.h (hunk 1 above)
Comment (Superjomn) [reply paddle/fluid/operators/tensorrt_engine_op.h]: This op should be called by single thread scenario

File: paddle/fluid/operators/tensorrt_engine_op.h (hunk 4 above)
Comment (Superjomn) [reply paddle/fluid/operators/tensorrt_engine_op.h]: Will create a c++ test when FC converter is merged latter.

**Pull request merged**